go/arrow/array/binary.go

```diff
@@ -56,9 +57,17 @@ func (a *Binary) Value(i int) []byte {
 	return a.valueBytes[a.valueOffsets[idx]:a.valueOffsets[idx+1]]
 }
 
-// ValueString returns the string at index i without performing additional allocations.
-// The string is only valid for the lifetime of the Binary array.
+// ValueString returns the string at index i
 func (a *Binary) ValueString(i int) string {
+	if a.IsNull(i) {
+		return NullValueStr
+	}
+	return base64.StdEncoding.EncodeToString(a.Value(i))
+}
+
+// ValueStr returns the string at index i without performing additional allocations.
+// The string is only valid for the lifetime of the Binary array.
+func (a *Binary) ValueStr(i int) string {
```

Comment:
Hmm i'd prefer to avoid the breaking change. Can we choose a different name for the function so we can avoid having to rename the `*Binary.ValueString` method?

Reply:
@zeroshade  I can do the other way around. `ValueStr` for the new function? WDYT?

Reply:
That's fine for me. Alternately, now that we have `GetOneForMarshal` publicly available, in theory with some trickery it should be pretty easy to leverage that and some strong typing with `String` methods we might not even need this `ValueStr` method but instead could simply just be able to `fmt.Sprint(GetOneForMarshal(...))` to get text. what do you think?

Reply:
Interesting. I prefer though to make it a public function `ValueStr` and the reverse `AppendFromValueStr` so it's easy to test it. I did call `GetOneForMarshal` in some of the `ValueStr` implementation where it made sense but I think in any case better to expose a proper function.

Reply:
I've updated. Not sure why the tests now fail though in the CI

Reply:
You missed a few spots for `arraymarshal`:
![image](https://user-images.githubusercontent.com/555095/231020740-14e46721-30ce-4419-992e-07362e607771.png)

Reply:
Ok seems I was missing a rebase. fixed now.